jobs/JGLOBAL_ATMOS_POST

```diff
@@ -1,46 +1,51 @@
 #! /usr/bin/env bash
 
-source "$HOMEgfs/ush/preamble.sh"
+# shellcheck source=ush/preamble.sh
+source "${HOMEgfs:?}/ush/preamble.sh"
 
 export RUN_ENVIR=${RUN_ENVIR:-"nco"}
 
 #############################
 # Source relevant config files
 #############################
 configs="base post"
-export EXPDIR=${EXPDIR:-$HOMEgfs/parm/config}
-config_path=${EXPDIR:-$PACKAGEROOT/gfs.${gfs_ver}/parm/config}
-for config in $configs; do
-    . $config_path/config.$config
+export EXPDIR=${EXPDIR:-${HOMEgfs}/parm/config}
+config_path=${EXPDIR:-${PACKAGEROOT}/gfs.${gfs_ver}/parm/config}
+for config in ${configs}; do
+    # shellcheck source-path=parm/config
+    . "${config_path}/config.${config}"
     status=$?
-    [[ $status -ne 0 ]] && exit $status
+    (( status != 0 )) && exit "${status}"
```

Comment:
Are you sure `(( )) && ` or `(( )) || ` works?
My understanding is that `(( ))` is used when performing arithmetic.  The result of the arithmetic will set the exit code (0 for true).

Reply:
Yes. (( )) evaluates in arithmetic context and can handle inequalities: https://mywiki.wooledge.org/ArithmeticExpression

Quick test:
```
hfe02 rename_post/ > if ((  2 > 0 )); then echo "hi"; fi
hi
hfe02 rename_post/ > if ((  0 > 2 )); then echo "hi"; fi                                                                                                  
hfe02 rename_post/ > (( 2 > 0 )) || echo "hi"
hfe02 rename_post/ > (( 0 > 2 )) || echo "hi"                                                                                                             
hi
```

Reply:
Are we testing here if the arithmetic works or evaluating the expression?
```
(( $status -ne 0 ))
```
returns the status of the arithmetic evaluation of the expression.  So even if the arithmetic was wrong, but evaluated incorrectly, the return status would be True.
```
[[ $status -ne 0 ]]
```
evaluates the expression and returns a True or False.

Reply:
Guess I'm not sure I see the difference unless there is a string where a number should be (since a string would be treated as 0 in an arithmetic context).

Reply:
I suppose it doesn't make a difference in the end.  Just be consistent and make it a rule.
```
(( 2*0 )) && echo "hi"
(( 2*1 )) && echo "hi"
hi
```
Not sure why the first is `False`.  It is a perfectly fine arithmetic expression.

Reply:
I would only use this form as a logical if it includes a comparison operator like the examples I gave. Really, we would never use `(( var = a + b ))` for math either, as it will return a non-zero exit code most of the time (use `var=$(( a + b ))` instead).